engine-cdi/src/main/java/org/camunda/bpm/engine/cdi/impl/ManagedJobExecutor.java

```diff
@@ -18,43 +18,49 @@
+
 import javax.enterprise.concurrent.ManagedExecutorService;
-import javax.enterprise.concurrent.ManagedThreadFactory;
+import org.camunda.bpm.engine.ProcessEngineException;
 import org.camunda.bpm.engine.impl.ProcessEngineImpl;
 import org.camunda.bpm.engine.impl.jobexecutor.JobExecutor;
-import org.camunda.bpm.engine.impl.jobexecutor.ThreadPoolJobExecutor;
 
 /**
- * {@link JobExecutor} implementation that utilises an application server's managed thread pool to acquire and execute jobs.
+ * {@link JobExecutor} implementation that utilises an application server's
+ * managed thread pool to acquire and execute jobs.
  */
-public class ManagedJobExecutor extends ThreadPoolJobExecutor {
+public class ManagedJobExecutor extends JobExecutor {
 
-    private ManagedExecutorService managedExecutorService;
-    private ManagedThreadFactory managedThreadFactory;
+  private ManagedExecutorService managedExecutorService;
 
-    /**
-     * Constructs a new ManagedJobExecutor with the provided {@link ManagedExecutorService} and {@link ManagedThreadFactory}.
-     */
-    public ManagedJobExecutor(final ManagedExecutorService managedExecutorService, final ManagedThreadFactory managedThreadFactory) {
-        this.managedExecutorService = managedExecutorService;
-        this.managedThreadFactory = managedThreadFactory;
-    }
+  /**
+   * Constructs a new ManagedJobExecutor with the provided
+   * {@link ManagedExecutorService}
+   */
+  public ManagedJobExecutor(final ManagedExecutorService managedExecutorService) {
+    this.managedExecutorService = managedExecutorService;
+  }
 
-    @Override
-    public void executeJobs(List<String> jobIds, ProcessEngineImpl processEngine) {
-        try {
-            managedExecutorService.execute(getExecuteJobsRunnable(jobIds, processEngine));
-        } catch (RejectedExecutionException e) {
-            logRejectedExecution(processEngine, jobIds.size());
-            rejectedJobsHandler.jobsRejected(jobIds, processEngine, this);
-        }
+  @Override
+  public void executeJobs(List<String> jobIds, ProcessEngineImpl processEngine) {
+    try {
+      managedExecutorService.execute(getExecuteJobsRunnable(jobIds, processEngine));
+    } catch (RejectedExecutionException e) {
+      logRejectedExecution(processEngine, jobIds.size());
+      rejectedJobsHandler.jobsRejected(jobIds, processEngine, this);
     }
+  }
 
-    @Override
-    protected void startJobAcquisitionThread() {
-        if (jobAcquisitionThread == null) {
-            jobAcquisitionThread = managedThreadFactory.newThread(acquireJobsRunnable);
-            jobAcquisitionThread.start();
-        }
+  @Override
+  protected void startExecutingJobs() {
+    try {
+      managedExecutorService.execute(acquireJobsRunnable);
+    } catch (Exception e) {
+      throw new ProcessEngineException("Could not schedule AcquireJobsRunnable for execution.", e);
     }
+  }
+
+  @Override
+  protected void stopExecutingJobs() {
+    // nothing to do
```

Comment:
As pointed out in #533 , I think there should be some handling here.

Reply:
That was my first instinct as well. As far as I can tell, this is however not necessary.
I understand the lifecycle as follows:
1. the JobExecutor is started via `JobExecutor#start`
   1. the acquisition runnable is created but not started
   1. `ManagedJobExecutor#startExecutingJobs` is called and instructs the `ManagedExecutorService` to start the acquisition runnable (it allocates a Thread for that, it keeps on running until we stop it)
1. the JobExecutor is stopped via `JobExecutor#shutdown`
   1. the acquisition runnable is stopped and thereby marked as `interrupted` in a flag
   1. the `SequentialJobAcquisitionRunnable#runAcquisition` loop running in `SequentialJobAcquisitionRunnable#run` is interrupted, the acquisition runnable stops
   1. the `ManagedExecutorService` frees the Thread it used for the acquisition runnable because that runnable stopped and returns the Thread to its pool of idle Threads

If I understand that lifecycle correctly, there shouldn't be anything left for us to do here. Please correct me if I missed something here.

Reply:
Yes, I see what you mean. It seems that from the behaviour of the shutdown of the JobExecutor itself , the runnable is already stopped, so there is no need to implement anything specific for the stopExecutingJobs method.

Reply:
Thanks for having a look at this 👍